config/routes.rb

```diff
@@ -14,9 +14,23 @@
   resources :categories, only: %i[index show] do
     resources :books, only: %i[index show]
   end
+
+  resources :checkouts
+  resources :coupons
+  resources :carts, only: %i[destroy show update] do
+    resources :coupons
+  end
   resources :books, only: [] do
     resources :reviews
   end
+
+  resources :orders, only: %i[index show new]
+  post 'line_items/:id/add' => 'line_items#add_quantity', as: 'line_item_add'
```

Comment:
make it as resources with included in block custom actions

Reply:
done